src/main/java/frc/robot/subsystems/drive/SwerveModuleIO.java

```diff
@@ -1,27 +1,54 @@
 package frc.robot.subsystems.drive;
 
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+
 public interface SwerveModuleIO {
-    public void setDriveEncoderPosition(double position);
 
-    public double getDriveEncoderPosition();
+    public static class SwerveModuleIOStates {
+
+        public double driveVoltage = 0.0;
+        public double turnVoltage = 0.0;
+        public double driveVelocity = 0.0;
+        public double driveDesiredVelocity = 0.0;
+        public double turnAngle = 0.0;
+        public double desiredAngle = 0.0;
+        public double driveCurrent = 0.0;
+        public double turnCurrent = 0.0;
+        public double driveEncoderPos = 0.0;
+        public double turningEncoderPos = 0.0;
+
+    }
+
+    public static class DriveSubsystemStates {
```

Comment:
Should really be defined in DriveSubsystem.java, not in the SwerveModuleIO.java, I think?